hal/mbed_sleep_manager.c

```diff
@@ -162,6 +162,7 @@ void sleep_manager_lock_deep_sleep_internal(void)
     if (deep_sleep_lock == USHRT_MAX) {
         core_util_critical_section_exit();
         MBED_ERROR1(MBED_MAKE_ERROR(MBED_MODULE_HAL, MBED_ERROR_CODE_OVERFLOW), "DeepSleepLock overflow (> USHRT_MAX)", deep_sleep_lock);
+        return; // Handle overridden mbed_error(), i.e. when running sleep_manager tests.
```

Comment:
This return was added so while running tests, it would immediately return instead of continue (does not trap)?

Reply:
Yes, since I overridden `mbed_error()` for test purposes and now it does not halt, I had to add this return to prevent execution of following lines -- changing the value of `deep_sleep_lock` var.